src/main.c

```diff
@@ -188,48 +192,50 @@ version(void)
 void
 get_choices(void)
 {
-	char		*line, *description, *field_separators;
-	size_t		 line_size;
+	char		*description, *field_separators, *start, *stop;
 	ssize_t		 length;
 	struct choice	*choice;
 
 	if ((field_separators = getenv("IFS")) == NULL)
 		field_separators = " ";
 
+	input.size = BUFSIZ;
+	if ((input.v = malloc(input.size)) == NULL)
+		err(1, "malloc");
+	for (;;) {
+		if ((length = read(0, input.v + input.nmemb,
```

Comment:
Do you think replacing `0` with `STDIN_FILENO` would be make this easier to read?